global.R

```diff
@@ -237,3 +237,131 @@ roundFiveUp <- function(x, n) {
+  }
+
+  if (top == "center") {
+    top <- "50%"
+    ytransform <- "-50%"
+  } else if (is.numeric(top) && top >= 0) {
+    top <- paste0(top, "px")
+    ytransform <- "0"
+  } else {
+    stop("disconnectMessage: 'top' must be either an integer, or the string \"center\".", call. = FALSE)
+  }
+
+  htmltools::tagList(
+    htmltools::tags$script(
+      paste0(
+        "$(function() {",
+        "  $(document).on('shiny:disconnected', function(event) {",
+        "    $('#custom-disconnect-dialog').show();",
+        "    $('#ss-overlay').show();",
+        "  })",
+        "});"
+      )
+    ),
+    htmltools::tags$div(
+      id = "custom-disconnect-dialog",
+      style = "display: none !important;",
+
+      htmltools::tags$div(
+        id = "ss-connect-refresh",
+        htmltools::tags$p("Something went wrong! Try refreshing the page."),
+        htmltools::tags$a(id = "ss-reload-link", href = "#", onclick = "window.location.reload(true);")
+      ),
+
+      htmltools::tags$div(
+        id = "ss-connect-image",
+        style = "display: block !important;",
+
+        htmltools::tags$img(id = "ss-reload-image", src = "builder-duck.PNG"),
+        htmltools::tags$p("If this persists, please contact statistics.development@education.gov.uk with details of what you were trying to do.")
+      )
+    ),
+
+    htmltools::tags$div(id = "ss-overlay", style = "display: none;"),
+
+    htmltools::tags$head(htmltools::tags$style(
+      glue::glue(
+        .open = "{{", .close = "}}",
+
+        ## This hides the old message
+        "#ss-connect-dialog { display: none !important; }", # rsconnect
+        "#shiny-disconnected-overlay { display: none !important; }", # local
+
+        "#ss-overlay {
+             background-color: {{overlayColour}} !important;
+             opacity: {{overlayOpacity}} !important;
+             position: fixed !important;
+             top: 0 !important;
+             left: 0 !important;
+             bottom: 0 !important;
+             right: 0 !important;
+             z-index: 99998 !important;
+             overflow: hidden !important;
+             cursor: not-allowed !important;
+          }",
+
+        "#custom-disconnect-dialog {
+             background: {{background}} !important;
+             color: {{colour}} !important;
+             width: {{width}} !important;
+             transform: translateX(-50%) translateY({{ytransform}}) !important;
+             font-size: {{size}}px !important;
+             top: {{top}} !important;
+             position: fixed !important;
+             bottom: auto !important;
+             left: 50% !important;
+             padding: 0.8em 1.5em !important;
+             text-align: center !important;
+             height: auto !important;
+             opacity: 1 !important;
+             z-index: 99999 !important;
+             border-radius: 3px !important;
+             box-shadow: rgba(0, 0, 0, 0.3) 3px 3px 10px !important;
```

Comment:
Not sure there's enough !important 's in this function...

Reply:
Hah I know! That bit is just as is from the initial function - with the differences between local and rsconnect making it so much harder I didn't bother testing if I could remove any to see if it still worked